### PR TITLE
Use the same G and H generators in range proof and set membership proof. Allow TokenId and blsct::Message as a seed for range proof

### PR DESCRIPTION
**Pull request by gogoex**
The changes:
- Introduce Seed of type std::variant<TokenId, blsct::Message>
- Use Seed as an input instead of TokenId for range proof prove and verify functions
- RangeProof class has Seed instead of TokenId
- RangeProof allows de/serialization only when Seed is TokenId 
- Use the G and H generators derived from given Seed in range proof and set membership proof
- Use different source of entropy for phi and fiat shamir in set membership proof
- Update bls to the latest version

### CODE REVIEW

File: src/blsct/range_proof/bulletproofs/range_proof_with_transcript.cpp [hunk 1]
```diff
@@ -2,17 +2,20 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
+#include "blsct/range_proof/bulletproofs/range_proof.h"
```
Comment (mxaddict) [inline src/blsct/range_proof/bulletproofs/range_proof_with_transcript.cpp:5]: I think this include needs to be changed to to <> instead of ""

File: src/bls/mcl/src/asm/x86-64.S
Comment (mxaddict) [inline src/bls/mcl/src/asm/x86-64.S:1]: Why was this file renamed?

File: src/blsct/range_proof/bulletproofs/range_proof_with_transcript.cpp (hunk 1 above)
Comment (gogoex) [reply src/blsct/range_proof/bulletproofs/range_proof_with_transcript.cpp:5]: I will fix it.

Comment (aguycalled) [reply src/blsct/range_proof/bulletproofs/range_proof_with_transcript.cpp:5]: this line needs to be removed

File: src/blsct/range_proof/bulletproofs_plus/range_proof_logic.cpp [hunk 2]
```diff
@@ -496,7 +495,8 @@ template <typename T>
 bool RangeProofLogic<T>::Verify(
     const std::vector<RangeProof<T>>& proofs
 ) {
-    range_proof::Common<T>::ValidateProofsBySizes(proofs);
+    // TODO uncomment this if using bulletproofs+ in prod
+    //range_proof::Common<T>::ValidateProofsBySizes(proofs);
```
Comment (aguycalled) [inline src/blsct/range_proof/bulletproofs_plus/range_proof_logic.cpp:499]: why is the line commented?

File: src/blsct/range_proof/bulletproofs/range_proof_with_transcript.cpp (hunk 1 above)
Comment (gogoex) [reply src/blsct/range_proof/bulletproofs/range_proof_with_transcript.cpp:5]: Right. `range_proof.h` is already included from `range_proof_with_transcript.h`. I removed the line.

File: src/bls/mcl/src/asm/x86-64.S
Comment (gogoex) [reply src/bls/mcl/src/asm/x86-64.S:1]: I didn't notice that this file was renamed. I renamed the file to the original name.

**Review by mxaddict — APPROVED** [review]

**Review by aguycalled — APPROVED** [review]

**Review by aguycalled — APPROVED** [review]
